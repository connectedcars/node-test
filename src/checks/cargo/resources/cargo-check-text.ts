import { CargoMessage } from '../cargo-types'

export const cargoCheckFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused import: `std::io::BufReader`\n --> src/main.rs:3:5\n  |\n3 | use std::io::BufReader;\n  |     ^^^^^^^^^^^^^^^^^^\n  |\n  = note: `#[warn(unused_imports)]` on by default\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_imports)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'remove the whole `use` item',
          rendered: null,
          spans: [
            {
              byte_end: 52,
              byte_start: 29,
              column_end: 24,
              column_start: 1,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 3,
              line_start: 3,
              suggested_replacement: '',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 24,
                  highlight_start: 1,
                  text: 'use std::io::BufReader;'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_imports',
        explanation: null
      },
      level: 'warning',
      message: 'unused import: `std::io::BufReader`',
      spans: [
        {
          byte_end: 51,
          byte_start: 33,
          column_end: 23,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 3,
          line_start: 3,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 23,
              highlight_start: 5,
              text: 'use std::io::BufReader;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused import: `std::io::BufReader`\n --> src/main.rs:3:5\n  |\n3 | use std::io::BufReader;\n  |     ^^^^^^^^^^^^^^^^^^\n  |\n  = note: `#[warn(unused_imports)]` on by default\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_imports)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'remove the whole `use` item',
          rendered: null,
          spans: [
            {
              byte_end: 52,
              byte_start: 29,
              column_end: 24,
              column_start: 1,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 3,
              line_start: 3,
              suggested_replacement: '',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 24,
                  highlight_start: 1,
                  text: 'use std::io::BufReader;'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_imports',
        explanation: null
      },
      level: 'warning',
      message: 'unused import: `std::io::BufReader`',
      spans: [
        {
          byte_end: 51,
          byte_start: 33,
          column_end: 23,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 3,
          line_start: 3,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 23,
              highlight_start: 5,
              text: 'use std::io::BufReader;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused variable: `foo`\n  --> src/main.rs:15:11\n   |\n15 |       let foo = foo();\n   |           ^^^ help: if this is intentional, prefix it with an underscore: `_foo`\n   |\n   = note: `#[warn(unused_variables)]` on by default\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_variables)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'if this is intentional, prefix it with an underscore',
          rendered: null,
          spans: [
            {
              byte_end: 160,
              byte_start: 157,
              column_end: 14,
              column_start: 11,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 15,
              line_start: 15,
              suggested_replacement: '_foo',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 14,
                  highlight_start: 11,
                  text: '      let foo = foo();'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_variables',
        explanation: null
      },
      level: 'warning',
      message: 'unused variable: `foo`',
      spans: [
        {
          byte_end: 160,
          byte_start: 157,
          column_end: 14,
          column_start: 11,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 15,
          line_start: 15,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 14,
              highlight_start: 11,
              text: '      let foo = foo();'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused variable: `foo`\n  --> src/main.rs:15:11\n   |\n15 |       let foo = foo();\n   |           ^^^ help: if this is intentional, prefix it with an underscore: `_foo`\n   |\n   = note: `#[warn(unused_variables)]` on by default\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_variables)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'if this is intentional, prefix it with an underscore',
          rendered: null,
          spans: [
            {
              byte_end: 160,
              byte_start: 157,
              column_end: 14,
              column_start: 11,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 15,
              line_start: 15,
              suggested_replacement: '_foo',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 14,
                  highlight_start: 11,
                  text: '      let foo = foo();'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_variables',
        explanation: null
      },
      level: 'warning',
      message: 'unused variable: `foo`',
      spans: [
        {
          byte_end: 160,
          byte_start: 157,
          column_end: 14,
          column_start: 11,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 15,
          line_start: 15,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 14,
              highlight_start: 11,
              text: '      let foo = foo();'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused variable: `f`\n  --> src/main.rs:20:9\n   |\n20 |     let f = File::open("foo.txt");\n   |         ^ help: if this is intentional, prefix it with an underscore: `_f`\n\n',
      children: [
        {
          code: null,
          level: 'help',
          message: 'if this is intentional, prefix it with an underscore',
          rendered: null,
          spans: [
            {
              byte_end: 213,
              byte_start: 212,
              column_end: 10,
              column_start: 9,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 20,
              line_start: 20,
              suggested_replacement: '_f',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 10,
                  highlight_start: 9,
                  text: '    let f = File::open("foo.txt");'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_variables',
        explanation: null
      },
      level: 'warning',
      message: 'unused variable: `f`',
      spans: [
        {
          byte_end: 213,
          byte_start: 212,
          column_end: 10,
          column_start: 9,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 20,
          line_start: 20,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 10,
              highlight_start: 9,
              text: '    let f = File::open("foo.txt");'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused variable: `f`\n  --> src/main.rs:20:9\n   |\n20 |     let f = File::open("foo.txt");\n   |         ^ help: if this is intentional, prefix it with an underscore: `_f`\n\n',
      children: [
        {
          code: null,
          level: 'help',
          message: 'if this is intentional, prefix it with an underscore',
          rendered: null,
          spans: [
            {
              byte_end: 213,
              byte_start: 212,
              column_end: 10,
              column_start: 9,
              expansion: null,
              file_name: 'src/main.rs',
              is_primary: true,
              label: null,
              line_end: 20,
              line_start: 20,
              suggested_replacement: '_f',
              suggestion_applicability: 'MachineApplicable',
              text: [
                {
                  highlight_end: 10,
                  highlight_start: 9,
                  text: '    let f = File::open("foo.txt");'
                }
              ]
            }
          ]
        }
      ],
      code: {
        code: 'unused_variables',
        explanation: null
      },
      level: 'warning',
      message: 'unused variable: `f`',
      spans: [
        {
          byte_end: 213,
          byte_start: 212,
          column_end: 10,
          column_start: 9,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 20,
          line_start: 20,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 10,
              highlight_start: 9,
              text: '    let f = File::open("foo.txt");'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: value assigned to `foo` is never read\n  --> src/main.rs:29:9\n   |\n29 |     let mut foo = 0;\n   |         ^^^^^^^\n   |\n   = note: `#[warn(unused_assignments)]` on by default\n   = help: maybe it is overwritten before being read?\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_assignments)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'maybe it is overwritten before being read?',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_assignments',
        explanation: null
      },
      level: 'warning',
      message: 'value assigned to `foo` is never read',
      spans: [
        {
          byte_end: 355,
          byte_start: 348,
          column_end: 16,
          column_start: 9,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 29,
          line_start: 29,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 16,
              highlight_start: 9,
              text: '    let mut foo = 0;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: value assigned to `foo` is never read\n  --> src/main.rs:29:9\n   |\n29 |     let mut foo = 0;\n   |         ^^^^^^^\n   |\n   = note: `#[warn(unused_assignments)]` on by default\n   = help: maybe it is overwritten before being read?\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_assignments)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'help',
          message: 'maybe it is overwritten before being read?',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_assignments',
        explanation: null
      },
      level: 'warning',
      message: 'value assigned to `foo` is never read',
      spans: [
        {
          byte_end: 355,
          byte_start: 348,
          column_end: 16,
          column_start: 9,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 29,
          line_start: 29,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 16,
              highlight_start: 9,
              text: '    let mut foo = 0;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: value assigned to `bar` is never read\n  --> src/main.rs:33:5\n   |\n33 |     bar = foo;\n   |     ^^^\n   |\n   = help: maybe it is overwritten before being read?\n\n',
      children: [
        {
          code: null,
          level: 'help',
          message: 'maybe it is overwritten before being read?',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_assignments',
        explanation: null
      },
      level: 'warning',
      message: 'value assigned to `bar` is never read',
      spans: [
        {
          byte_end: 405,
          byte_start: 402,
          column_end: 8,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 33,
          line_start: 33,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 8,
              highlight_start: 5,
              text: '    bar = foo;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: value assigned to `bar` is never read\n  --> src/main.rs:33:5\n   |\n33 |     bar = foo;\n   |     ^^^\n   |\n   = help: maybe it is overwritten before being read?\n\n',
      children: [
        {
          code: null,
          level: 'help',
          message: 'maybe it is overwritten before being read?',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_assignments',
        explanation: null
      },
      level: 'warning',
      message: 'value assigned to `bar` is never read',
      spans: [
        {
          byte_end: 405,
          byte_start: 402,
          column_end: 8,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 33,
          line_start: 33,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 8,
              highlight_start: 5,
              text: '    bar = foo;'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused `std::result::Result` that must be used\n  --> src/main.rs:25:5\n   |\n25 |     f.read(&mut buf);\n   |     ^^^^^^^^^^^^^^^^^\n   |\n   = note: `#[warn(unused_must_use)]` on by default\n   = note: this `Result` may be an `Err` variant, which should be handled\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_must_use)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'note',
          message: 'this `Result` may be an `Err` variant, which should be handled',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_must_use',
        explanation: null
      },
      level: 'warning',
      message: 'unused `std::result::Result` that must be used',
      spans: [
        {
          byte_end: 336,
          byte_start: 319,
          column_end: 22,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 25,
          line_start: 25,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 22,
              highlight_start: 5,
              text: '    f.read(&mut buf);'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered:
        'warning: unused `std::result::Result` that must be used\n  --> src/main.rs:25:5\n   |\n25 |     f.read(&mut buf);\n   |     ^^^^^^^^^^^^^^^^^\n   |\n   = note: `#[warn(unused_must_use)]` on by default\n   = note: this `Result` may be an `Err` variant, which should be handled\n\n',
      children: [
        {
          code: null,
          level: 'note',
          message: '`#[warn(unused_must_use)]` on by default',
          rendered: null,
          spans: []
        },
        {
          code: null,
          level: 'note',
          message: 'this `Result` may be an `Err` variant, which should be handled',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'unused_must_use',
        explanation: null
      },
      level: 'warning',
      message: 'unused `std::result::Result` that must be used',
      spans: [
        {
          byte_end: 336,
          byte_start: 319,
          column_end: 22,
          column_start: 5,
          expansion: null,
          file_name: 'src/main.rs',
          is_primary: true,
          label: null,
          line_end: 25,
          line_start: 25,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 22,
              highlight_start: 5,
              text: '    f.read(&mut buf);'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered: 'warning: 6 warnings emitted\n\n',
      children: [],
      code: null,
      level: 'warning',
      message: '6 warnings emitted',
      spans: []
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    message: {
      rendered: 'warning: 6 warnings emitted\n\n',
      children: [],
      code: null,
      level: 'warning',
      message: '6 warnings emitted',
      spans: []
    }
  },
  {
    reason: 'compiler-artifact',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    profile: {
      opt_level: '0',
      debuginfo: 2,
      debug_assertions: true,
      overflow_checks: true,
      test: false
    },
    features: [],
    filenames: ['/checks-test/target/debug/deps/libcheck_test-3e8dea89f4957cbe.rmeta'],
    executable: null,
    fresh: false
  },
  {
    reason: 'compiler-artifact',
    package_id: 'checks-test 0.1.0 (path+file:///checks-test)',
    target: {
      kind: ['bin'],
      crate_types: ['bin'],
      name: 'checks-test',
      src_path: '/checks-test/src/main.rs',
      edition: '2018',
      doctest: false,
      test: true
    },
    profile: {
      opt_level: '0',
      debuginfo: 2,
      debug_assertions: true,
      overflow_checks: true,
      test: true
    },
    features: [],
    filenames: ['/checks-test/target/debug/deps/libcheck_test-b71ac6c54ee2a77f.rmeta'],
    executable: null,
    fresh: false
  },
  {
    reason: 'build-finished',
    success: true
  }
]
