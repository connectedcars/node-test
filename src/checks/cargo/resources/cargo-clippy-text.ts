import { CargoMessage } from '../cargo-types'

export const cargoClippyFailedOutput: CargoMessage[] = [
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered:
        'error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false\n   --> can/src/can_session.rs:391:12\n    |\n391 |         if 100 > i32::MAX {}\n    |            ^^^^^^^^^^^^^^\n    |\n    = note: `-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`\n    = help: because `i32::MAX` is the maximum value for this type, this comparison is always false\n    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons\n\n',
      children: [
        {
          children: [],
          code: null,
          level: 'note',
          message: '`-D clippy::absurd-extreme-comparisons` implied by `-D clippy::all`',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message: 'because `i32::MAX` is the maximum value for this type, this comparison is always false',
          rendered: null,
          spans: []
        },
        {
          children: [],
          code: null,
          level: 'help',
          message:
            'for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons',
          rendered: null,
          spans: []
        }
      ],
      code: {
        code: 'clippy::absurd_extreme_comparisons',
        explanation: null
      },
      level: 'error',
      message:
        'this comparison involving the minimum or maximum element for this type contains a case that is always true or always false',
      spans: [
        {
          byte_end: 13338,
          byte_start: 13324,
          column_end: 26,
          column_start: 12,
          expansion: null,
          file_name: 'can/src/can_session.rs',
          is_primary: true,
          label: null,
          line_end: 391,
          line_start: 391,
          suggested_replacement: null,
          suggestion_applicability: null,
          text: [
            {
              highlight_end: 26,
              highlight_start: 12,
              text: '        if 100 > i32::MAX {}'
            }
          ]
        }
      ]
    }
  },
  {
    reason: 'compiler-message',
    package_id: 'can 0.1.0 (path+file:///app/can)',
    target: {
      kind: ['lib'],
      crate_types: ['lib'],
      name: 'can',
      src_path: '/app/can/src/lib.rs',
      edition: '2018',
      doctest: true
    },
    message: {
      rendered: 'error: aborting due to previous error\n\n',
      children: [],
      code: null,
      level: 'error',
      message: 'aborting due to previous error',
      spans: []
    }
  }
]
