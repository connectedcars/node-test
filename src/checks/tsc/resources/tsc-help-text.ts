export const tscSuccesfulOutput = ``
export const tscErrorOutput = `src/common.ts(4,88): error TS1128: Declaration or statement expected.`
export const tscMultiErrorOutput = `src/common.ts(4,88): error TS1128: Declaration or statement expected.
src/index.ts(29,14): error TS1109: Expression expected.`
export const tscCommandErrorOutput = `error TS5053: Option 'noEmit' cannot be specified with option 'incremental'`
