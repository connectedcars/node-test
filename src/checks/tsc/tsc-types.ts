interface TscData {
  file: string
  line: number
  col: number
  errorCode: string
  message: string
}
