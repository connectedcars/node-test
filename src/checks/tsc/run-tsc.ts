import { splitLines } from '../../common'
import { RunProcess } from '../../unix/run-process'
import { TscData } from './tsc-types'

function matchToObj(match: RegExpExecArray): TscData {
  return {
    file: match[1],
    line: parseInt(match[2], 10),
    col: parseInt(match[3], 10),
    errorCode: match[4],
    message: match[5] ? match[5] : ''
  }
}

export function parseTsc(output: string): TscData[] {
  const messages = splitLines(output)

  const tscErrorRegex = /^((?:[\w-]+\/)*[\w.-]+)\((\d+),(\d+)\): error (TS\d+): (.*)/

  const errors: TscData[] = []

  for (const msg of messages) {
    const matched = tscErrorRegex.exec(msg)
    if (matched !== null) {
      errors.push(matchToObj(matched))
    } else if (errors.length > 0) {
      // this is a continuation of the previous error
      errors[errors.length - 1].message += '\n' + msg
    }
  }

  return errors
}

export async function runTsc(extraArgs: string[] = []): Promise<TscData[]> {
  const cmd = new RunProcess('tsc', [...extraArgs, '--pretty', 'false', '--noErrorTruncation', '--noEmit'])
  const data: Buffer[] = []
  cmd.stdout?.on('data', (chunk: Buffer) => {
    data.push(chunk)
  })
  await cmd.waitForStarted()
  const { code } = await cmd.waitForExit()
  if (code != 0) {
    throw new Error(`tsc failed: ${data.toString()}`)
  }
  const compileResult = Buffer.concat(data).toString('utf8')
  return parseTsc(compileResult)
}
