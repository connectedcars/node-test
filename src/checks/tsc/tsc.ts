import { CheckConversionError, CheckOutput, CheckRunCompleted, CheckRunConclusion } from '../checks-common'
import { TscData } from './tsc-types'

export interface TscInput {
  data: TscData[]
  sha: string
}

export function tscCheck({ data, sha }: TscInput): CheckRunCompleted {
  try {
    let conclusion: CheckRunConclusion = 'success'
    const output: CheckOutput = {
      summary: 'No problems found',
      title: 'No problems found',
      annotations: []
    }

    if (data && data.length > 0) {
      conclusion = 'failure'
      output.summary = `${data.length} error(s) found`
      output.title = output.summary
      output.annotations = data.map(err => {
        const relPath = err.file
        return {
          start_line: err.line,
          end_line: err.line,
          annotation_level: 'failure',
          message: err.message,
          path: relPath,
          raw_details: JSON.stringify(err, null, 2)
        }
      })
    }

    return {
      name: 'tsc',
      head_sha: sha,
      conclusion,
      status: 'completed',
      completed_at: new Date().toISOString(),
      output
    }
  } catch (e) {
    throw new CheckConversionError('tsc', { data }, e as Error)
  }
}
