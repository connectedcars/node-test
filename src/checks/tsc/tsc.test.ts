/* tsc-disable @typescript-tsc/camelcase */
import { tscErrorOutput, tscMultiErrorOutput, tscSuccesfulOutput } from './resources/tsc-help-text'
import { parseTsc } from './run-tsc'
import { tscCheck } from './tsc'

describe('checks/tsc', () => {
  it('converts successful tsc', () => {
    const data = parseTsc(tscSuccesfulOutput)
    const result = tscCheck({
      data,
      sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88'
    })
    expect(result).toStrictEqual({
      name: 'tsc',
      head_sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88',
      conclusion: 'success',
      status: 'completed',
      completed_at: expect.stringMatching(/^\d{4}/),
      output: {
        title: 'No problems found',
        summary: 'No problems found',
        annotations: []
      }
    })
  })

  it('converts successful failed tsc', () => {
    const data = parseTsc(tscErrorOutput)
    const result = tscCheck({
      data,
      sha: 'c61a4ae014360e064eb2a9f76c8a6a55d05e5b88'
    })
    expect(result).toMatchSnapshot({
      completed_at: expect.stringMatching(/^\d{4}/)
    })
  })

  it('converts successful failed tsc with multi errors', () => {
    const data = parseTsc(tscMultiErrorOutput)
    const result = tscCheck({
      data,
      sha: 'abcdefg'
    })
    expect(result).toMatchSnapshot({
      completed_at: expect.stringMatching(/^\d{4}/)
    })
  })
})
