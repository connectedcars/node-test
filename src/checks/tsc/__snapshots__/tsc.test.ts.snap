// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks/tsc converts successful failed tsc 1`] = `
Object {
  "conclusion": "failure",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "end_line": 4,
        "message": "Declaration or statement expected.",
        "path": "src/common.ts",
        "raw_details": "{
  \\"file\\": \\"src/common.ts\\",
  \\"line\\": 4,
  \\"col\\": 88,
  \\"errorCode\\": \\"TS1128\\",
  \\"message\\": \\"Declaration or statement expected.\\"
}",
        "start_line": 4,
      },
    ],
    "summary": "1 error(s) found",
    "title": "1 error(s) found",
  },
}
`;

exports[`checks/tsc converts successful failed tsc with multi errors 1`] = `
Object {
  "conclusion": "failure",
  "output": Object {
    "annotations": Array [
      Object {
        "annotation_level": "failure",
        "end_line": 4,
        "message": "Declaration or statement expected.",
        "path": "src/common.ts",
        "raw_details": "{
  \\"file\\": \\"src/common.ts\\",
  \\"line\\": 4,
  \\"col\\": 88,
  \\"errorCode\\": \\"TS1128\\",
  \\"message\\": \\"Declaration or statement expected.\\"
}",
        "start_line": 4,
      },
      Object {
        "annotation_level": "failure",
        "end_line": 29,
        "message": "Expression expected.",
        "path": "src/index.ts",
        "raw_details": "{
  \\"file\\": \\"src/index.ts\\",
  \\"line\\": 29,
  \\"col\\": 14,
  \\"errorCode\\": \\"TS1109\\",
  \\"message\\": \\"Expression expected.\\"
}",
        "start_line": 29,
      },
    ],
    "summary": "2 error(s) found",
    "title": "2 error(s) found",
  },
}
`;
